internal/apps/agenda/restrict.go

```diff
@@ -67,3 +68,56 @@ func Restrict(r restricter.HasPermer) restricter.ElementFunc {
 		return element, nil
 	}
 }
+
+// RestrictListOfSpeakers restricts the list of speakers: The field "note" is
+// removed for every speaker, if the user is not a manager, the speaker user or
+// the config agenda_list_of_speakers_speaker_note_for_everyone is set.
+func RestrictListOfSpeakers(r restricter.HasPermer) restricter.ElementFunc {
+	return func(uid int, element json.RawMessage) (json.RawMessage, error) {
+		if !r.HasPerm(uid, CanSeeListOfSpeakers) {
+			return nil, nil
+		}
+
+		var notesForEveryone bool
+		if err := r.ConfigValue("agenda_list_of_speakers_speaker_note_for_everyone", &notesForEveryone); err != nil {
+			return nil, fmt.Errorf("getting agenda_list_of_speakers_speaker_note_for_everyone: %w", err)
+		}
+
+		if notesForEveryone || r.HasPerm(uid, pCanManageListOfSpeakers) {
+			return element, nil
+		}
+
+		// Delete each speaker's note for speakers which user is not the one to restrict for.
+		var los map[string]json.RawMessage
+		if err := json.Unmarshal(element, &los); err != nil {
+			return nil, fmt.Errorf("decoding list of speakers: %w", err)
+		}
+		var speakers []map[string]json.RawMessage
+		if err := json.Unmarshal(los["speakers"], &speakers); err != nil {
+			return nil, fmt.Errorf("decoding speakers: %w", err)
+		}
+
+		for _, speaker := range speakers {
+			var userId int
+			if err := json.Unmarshal(speaker["user_id"], &userId); err != nil {
+				return nil, fmt.Errorf("decoding speaker: %w", err)
```

Comment:
Each error message in a function should be different. In other case, it is not possible to see, which line was the problem

Reply:
It is, but subtle: `speaker` vs `speakers`

Reply:
Maybe I need glasses